lib/Amp/Tests/Process/ProcessBuilderTest.php

```diff
@@ -22,28 +22,33 @@ public function testBuildProcess(): void
     public function testDoesNotMergeEnvByDefaultWhenEnvVarsPassed(): void
     {
         putenv('FOO='.self::PARENT_PROCESS_ENV_VAR);
-        $process = ProcessBuilder::create(['export'])->env(['ENV'=> 'myenvvar'])->build();
+        $process = $this->makeEnvDumpProcess()->env(['ENV'=> 'myenvvar'])->build();
         $pid = wait($process->start());
-        $exitCode = wait($process->join());
-        self::assertEquals(0, $exitCode);
         /** @var string $out @phpstan-ignore-next-line */
         $out = wait(buffer($process->getStdout()));
         self::assertStringContainsString('myenvvar', $out);
         self::assertStringNotContainsString(self::PARENT_PROCESS_ENV_VAR, $out);
+        $exitCode = wait($process->join());
+        self::assertEquals(0, $exitCode);
         putenv('FOO');
     }
 
     public function testInheritsWhenInstructedToWhenEnvVarsPassed(): void
     {
         putenv('FOO='.self::PARENT_PROCESS_ENV_VAR);
-        $process = ProcessBuilder::create(['export'])->env(['ENV'=> 'myenvvar'])->mergeParentEnv()->build();
+        $process = $this->makeEnvDumpProcess()->env(['ENV'=> 'myenvvar'])->mergeParentEnv()->build();
         $pid = wait($process->start());
-        $exitCode = wait($process->join());
-        self::assertEquals(0, $exitCode);
         /** @var string $out @phpstan-ignore-next-line */
         $out = wait(buffer($process->getStdout()));
         self::assertStringContainsString('myenvvar', $out);
         self::assertStringContainsString(self::PARENT_PROCESS_ENV_VAR, $out);
+        $exitCode = wait($process->join());
+        self::assertEquals(0, $exitCode);
         putenv('FOO');
     }
+
+    private function makeEnvDumpProcess(): ProcessBuilder
+    {
+        return ProcessBuilder::create([PHP_BINARY, '-r', 'var_dump($_ENV);']);
```

Comment:
confused why this is necesary when we use `mergeParentEnv` - do processes not receive env vars on windows?

Reply:
Why don't we use the php `[getenv](https://www.php.net/manual/en/function.getenv.php)`? Since php8 it allows the parameter to be null and return all env vars.

Reply:
They do, but I'm afraid I don't really understand the question. ProcessBuilder (and Process) override that behavior for some reason, and this is a test for it.

Reply:
ok i see, ignore me